modules/ardrivo/test/String.cxx

```diff
@@ -0,0 +1,241 @@
+#include <catch2/catch.hpp>
+#include "WString.h"
+#include <string_view>
+
+using Catch::Contains;
+using Catch::EndsWith; 
+
+//Insufficient
+TEST_CASE("Test compare a string to another" "[compareTo]")
+{
+	String test = "test";
+	String test2 = "test";
+	String test3 = "test";
+	String test4 = "est";	
+	String test5 = "Test";
+	REQUIRE(test.compareTo(test2) == 0);
+	REQUIRE(test3.compareTo(test4) == 1);
+	REQUIRE(test5.compareTo(test) == -1);
+}
+
+TEST_CASE("Tests whether or not a String starts with the characters of another String" "[startsWith]")
+{
+	String test = "test";
+	String test2 = "test";
+	String test3 = "est";
+	String test4 = "test";	
+	REQUIRE(test.startsWith(test2));
+	REQUIRE(!test3.startsWith(test4));
+}
+
+TEST_CASE("Tests whether or not a String ends with the characters of another String" "[endsWith]")
+{
+	String test = "test";
+	String test2 = "test";
+	String test3 = "est";
+	String test4 = "test";
+	REQUIRE(test.startsWith(test2));
+	REQUIRE(!test3.startsWith(test4));
+}
+
+TEST_CASE("Copies the String�s characters to the supplied buffer" "[getBytes]")
+{
+	using Catch::Contains; 
+	String test = "test";
+	byte arr[5];
+	test.getBytes(arr, test.length());	
+	const char *p = reinterpret_cast<const char*>(arr);
+	REQUIRE_THAT(p, Contains("test"));
+}
+
+TEST_CASE("Locates a character or String within another String" "[indexOf]")
+{
+	String test = "test";
+	String test2 = "test";
+	String test3 = "est";
+	REQUIRE(test.indexOf(test2) == 0);
+	REQUIRE(test.indexOf(test3) == 1);
+}
+
+TEST_CASE("Modify in place a String removing chars from the provided index to the end of the String or from the provided index to index plus count" "[remove]")
+{
+	using Catch::EndsWith; 
+	String test = "test";
+	char arr[4];
+	test.remove(3);
+	strcpy(arr, test.c_str());
+	REQUIRE_THAT(arr, EndsWith("tes"));
+}
+
+TEST_CASE("Check if string gets swapped" "[replace]")
+{
+	using Catch::EndsWith; 
+	String test = "ab";
+	String test2 = "cd";
+	test.replace(test, test2);
+	char arr[3];
+	strcpy(arr, test.c_str());	
+	REQUIRE_THAT(arr, EndsWith( "cd" ));
+}
+
+//Crap does not work without std::string
+TEST_CASE("Test if string is given a new capacity" "[reserve]")
+{ 
+	String cap = "";
+	REQUIRE(cap.length() == 0);
+
+	cap.reserve(25);
+	//REQUIRE(cap.length() == 63);
+}
+
+TEST_CASE("Sets a character of the String" "[setCharAt]")
+{
+	using Catch::EndsWith; 
+	String test = "Hejsan";
+	test.setCharAt(1, '\x84');
+	char arr[7];
+	strcpy(arr, test.c_str());
+	REQUIRE_THAT(arr, EndsWith("H\x84jsan"));
+}
+
+TEST_CASE("Get a substring of a String" "[subsring]")
+{	
+	using Catch::Contains; 
+	String test = "test";
+	String newString = test.substring(2);
+	char arr[3];
+	strcpy(arr, newString.c_str());
+	REQUIRE_THAT(arr, Contains("st"));
```

Comment:
cant you directly do `REQUIRE_THAT(newString.c_str(), Catch::Constrains("st"));`? (seen it multiple times)